packages/iota/src/validation/outputs/common.ts

```diff
@@ -0,0 +1,347 @@
+
+    return mergeValidationResults(...results);
+}
+
+/**
+ * Validate the amount is greater than zero.
+ * @param amount The amount to validate.
+ * @returns The validation result.
+ */
+function validateAmountIsGreaterThanZero(amount: string, outputName: string): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (bigInt(amount).leq(bigInt.zero)) {
+        result = failValidation(result, `${outputName} output amount field must be larger than zero.`);
+    }
+
+    return result;
+}
+
+/**
+ * Validate the amount is lesser than maximum token supply.
+ * @param amount The amount to validate.
+ * @returns The validation result.
+ */
+function validateAmountIsLesserThanMaxSupply(amount: string, tokenSupply: string, outputName: string): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (bigInt(amount).gt(tokenSupply)) {
+        result = failValidation(result, `${outputName} output amount field must not be larger than max token supply.`);
+    }
+
+    return result;
+}
+
+/**
+ * Validate the count is within allowed boundries.
+ * @param outputName The name of the output to use in the error message.
+ * @param elementName The name of the validation subject to use in the error message.
+ * @param count The number to validate.
+ * @param max The maximum allowed value.
+ * @param min The maximum allowed value.
+ * @returns The validation result.
+ */
+//Basic
+// It must hold true that 1 ≤ Unlock Conditions Count ≤ 4.
+// It must hold true that 0 ≤ Features Count ≤ 3.
+//Alias
+// It must hold true that Unlock Conditions Count = 2.
+// It must hold true that 0 ≤ Features Count ≤ 2.
+// It must hold true that 0 ≤ Immutable Features Count ≤ 2.
+// Foundry
+// It must hold true that Unlock Conditions Count = 1.
+// It must hold true that 0 ≤ Features Count ≤ 1.
+// It must hold true that 0 ≤ Immutable Features Count ≤ 1.
+// Nft
+// It must hold true that 1 ≤ Unlock Conditions Count ≤ 4.
+// It must hold true that 0 ≤ Features Count ≤ 3.
+// It must hold true that 0 ≤ Immutable Features Count ≤ 2.
+function validateCount(outputName: string, elementName: string, count: number, max: number, min: number = 0): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (count < min || count > max ) {
+        const message = min === max ?
+            `${outputName} output ${elementName} count must be equal to ${max}.` :
+            `${outputName} output ${elementName} count must be between ${min} and ${max}.`;
+
+        result = failValidation(result, message);
+    }
+
+    return result;
+}
+
+/**
+ * Validate that the unlock conditions define only allowed types.
+ * @param unlockConditions The amount to validate.
+ * @param outputType The type of the output.
+ * @returns The validation result.
+ */
+function validateUnlockConditionsDefineAllowedTypes(unlockConditions: UnlockConditionTypes[], outputType: number, outputName: string): IValidationResult {
```

Comment:
I would drop the "Define" word from the name for this and methods bellow.
`validateUnlockConditionAllowedTypes`
`validateFeatureAllowedTypes`
...
And put the `outputType` parameter first, as that is the key piece from where the validation starts. As in, first we need to know for that type we are validating, then we need the unlockConditions to compare to the allowed ones.